src/System.Management.Automation/engine/runtime/CompiledScriptBlock.cs

```diff
@@ -1525,6 +1525,14 @@ private static bool WriteScriptBlockToLog(ScriptBlock scriptBlock, int segment,
                 // So, replace null characters with the Unicode `SYMBOL FOR NULL`
                 // We don't just remove the characters to preserve the fact that a null character was there.
                 textToLog = textToLog.Replace('\u0000', '\u2400');
```

Comment:
Do we need to handle `null` the same way for Linux and macOS?

Reply:
yes